### PR TITLE
Replace the Qt GUI with a React web GUI

### PR DESCRIPTION
**Pull request by egbertbouman**
This PR replaces the Qt GUI with a React web GUI (https://github.com/Tribler/tribler/issues/7736). Once merged, Tribler will just use 1 process, and will offer only a system tray icon to either open the GUI in the default web browser, or close the application.

The GUI itself is fully functional and takes into consideration that, in the future, users may want to run this on a server: .torrents are uploaded to the core and there is a `FileEndpoint` to browse the server's directory structure remotely.

There's still a lot of work to be done (e.g., the build scripts), but I'll do this in separate PRs.

### CODE REVIEW

**Comment by egbertbouman** [comment]
validate

File: src/run_tribler.py [hunk 1]
```diff
@@ -57,17 +62,44 @@ def main() -> None:
...
-        run_gui(api_port, api_key, root_state_dir)
+    await session.shutdown_event.wait()
+    await session.shutdown()
+    icon.stop()
+    logger.info("Tribler shutdown completed")
 
 
 if __name__ == "__main__":
-    main()
+    asyncio.run(main())
```
Comment (qstokkink) [inline src/run_tribler.py:104]: Won't this ruin the `asyncio.set_event_loop(asyncio.SelectorEventLoop())` on Windows?

Comment (egbertbouman) [reply src/run_tribler.py:104]: You're right. I'll move the `set_event_loop` call to just before calling `run`.

File: src/tribler/core/libtorrent/restapi/torrentinfo_endpoint.py [hunk 2]
```diff
@@ -219,3 +220,31 @@ async def get_torrent_info(self, request: Request) -> RESTResponse:  # noqa: C90
...
+        """
+        Return metainfo from a torrent found at a provided .torrent file.
+        """
+        tdef = TorrentDef.load_from_memory(await request.read())
+        infohash = tdef.get_infohash()
+
+        # Check if the torrent is already in the downloads
+        download = self.download_manager.downloads.get(infohash)
+        metainfo_request = self.download_manager.metainfo_requests.get(infohash, [None])[0]
+        requesting_metainfo = download == metainfo_request
```
Comment (qstokkink) [inline src/tribler/core/libtorrent/restapi/torrentinfo_endpoint.py]: This look similar to https://github.com/qstokkink/TriblerExperimental/pull/74/files#diff-40175f3551408276c12bea66ac0ed6672b8290e87d1fa1c8f483c04211f1e623R210-R212 Could you double-check if this needs the same fix?

Comment (egbertbouman) [reply src/tribler/core/libtorrent/restapi/torrentinfo_endpoint.py]: Oops, I fixed it on the main branch, but not on my own branch.

File: src/tribler/core/user_activity/manager.py [hunk 3]
```diff
@@ -41,12 +41,11 @@ def __init__(self, task_manager: TaskManager, session: Session, max_query_histor
         self.task_manager.register_task("Check preferable", self.check_preferable,
                                         interval=session.config.get("user_activity/health_check_interval"))
 
-    def on_query_results(self, data: dict) -> None:
```
Comment (qstokkink) [inline src/tribler/core/user_activity/manager.py:44]: If `query` is always a `str`, the next `if query is None:` can be removed.

**Review by qstokkink — CHANGES_REQUESTED** [review]
Please change this one minor thing.

Edit: and the [other thing above too](https://github.com/qstokkink/TriblerExperimental/pull/82#discussion_r1646195266) then.

File: src/tribler/core/user_activity/manager.py (hunk 3 above)
Comment (qstokkink) [reply src/tribler/core/user_activity/manager.py:44]: (this comment should be on the addition, not on the removal - woops)

Comment (egbertbouman) [reply src/tribler/core/user_activity/manager.py:44]: I basically made the typing up, thinking a search query without a query didn't make sense. Having said that, I can't really guarantee that's not somehow magically needed. The check was there before.
So, I'll either fix the typing or remove the check. Which do you prefer?

Comment (qstokkink) [reply src/tribler/core/user_activity/manager.py:44]: Like you said, I think it is sensible to change this to a strict `str`. For this reason, I'd prefer removing the check.

Comment (egbertbouman) [reply src/tribler/core/user_activity/manager.py:44]: Done. There were even tests for it, which I also removed.

**Comment by egbertbouman** [comment]
validate

**Review by qstokkink — APPROVED** [review]

**Pull request merged**